dali/imgcodec/decoders/nvjpeg2k/nvjpeg2k_test.cc

```diff
@@ -144,29 +165,59 @@ class NvJpeg2000DecoderTest : public NumpyDecoderTestBase<GPUBackend, OutputType
     auto decoded = this->Decode(make_span(in), this->GetParams(), make_span(rois));
     for (size_t i = 0; i < batch_size; i++) {
       auto ref = this->ReadReferenceFrom(data[i].ref_path);
-      this->AssertEqualSatNorm(decoded[i], ref);
+      this->AssertEqual(decoded[i], ref, eps);
     }
   }
 };
 
-using DecodeOutputTypes = ::testing::Types<uint8_t>;
+using DecodeOutputTypes = ::testing::Types<uint8_t, int16_t, float>;
 TYPED_TEST_SUITE(NvJpeg2000DecoderTest, DecodeOutputTypes);
 
 TYPED_TEST(NvJpeg2000DecoderTest, DecodeSingle) {
   for (const auto &name : images)
-    this->RunSingleTest(ImageTestingData{name});
+    this->RunTest(from_regular_file(name));
 }
 
 TYPED_TEST(NvJpeg2000DecoderTest, DecodeSingleRoi) {
   for (const auto &[name, roi] : roi_images)
-    this->RunSingleTest(ImageTestingData{name, roi});
+    this->RunTest(from_regular_file(name, roi));
 }
 
 TYPED_TEST(NvJpeg2000DecoderTest, DecodeBatchSingleThread) {
   std::vector<ImageTestingData> data;
   for (const auto &name : images)
-    data.emplace_back(name);
-  this->RunBatchTest(data);
+    data.push_back(from_regular_file(name));
+  this->RunTest(data);
+}
+
+TYPED_TEST(NvJpeg2000DecoderTest, 5BitImage) {
+  using OutputType = decltype(this->output_type_obj);
+  float eps;
+  if (std::is_floating_point<OutputType>::value)
+    eps = 0.07;
+  else if (std::is_same<OutputType, uint8_t>::value)
+    eps = 16.05;
+  else if (std::is_same<OutputType, int16_t>::value)
+    eps = 256 * 16.05;
+  else
+    assert(false);
+
+  this->RunTest(from_custom_bitdepth_file(bitdepth_converted_imgname, 5), eps);
+}
+
+TYPED_TEST(NvJpeg2000DecoderTest, 12BitImage) {
+  using OutputType = decltype(this->output_type_obj);
+  float eps;
+  if (std::is_floating_point<OutputType>::value)
+    eps = 0.01;
+  else if (std::is_same<OutputType, uint8_t>::value)
```

Comment:
std::is_same_v

Reply:
As an alternative you can instantiate any traits as it is _boolean-testable_ eg. ``std::is_same<T, int>{}``